build-common/src/org/jetbrains/kotlin/incremental/KotlinClassInfo.kt

```diff
@@ -152,8 +164,10 @@ private class InlineFunctionsAndAccessorsClassVisitor(
     }
 
     override fun visitMethod(access: Int, name: String, desc: String, signature: String?, exceptions: Array<out String>?): MethodVisitor? {
-        if (access and Opcodes.ACC_PRIVATE == Opcodes.ACC_PRIVATE) return null
-
+        // Note: Do not filter out private methods here because for some reason, the visibility of a Kotlin function does not always match
+        // the visibility of the corresponding JVM method (e.g., in `kotlin/CharCode.kt`, inline function `Char(code: Int): Char` is public,
+        // but the corresponding JVM method in `kotlin/CharCodeKt.class` is private).
```

Comment:
This is in fact expected with "kotlin.internal.InlineOnly", see `kotlin/compiler/frontend/src/org/jetbrains/kotlin/resolve/inline/inlineOnly.kt`.

Reply:
Nice, I've added that link to the comment.